scripts/tests/run_tv_casting_test.py

```diff
@@ -187,12 +213,12 @@ def extract_device_info_from_tv_casting_app(tv_casting_app_info: Tuple[subproces
         linux_tv_casting_app_log_file.flush()
 
         if 'device Name' in line:
-            device_name = extract_value_from_string(line)
+            device_name = extract_value_from_string(line, 'device Name', casting_state, log_paths)
```

Comment:
this repeats `device Name` in two places. I wonder if this should become:

```py
if value := extract_value_from_log_line(line, 'device name', casting_state, log_paths):
    device_name = value
```

and same below.
I also wonder about passing casting state and log paths every single time. It seems it would be more natural to have a `log_value_extractor = LogValueExtractor(casting_state, log_paths)` and then

```py
if value := log_value_extractor.extract_from(line, 'device Name'):
    device_name = value
```

Judgement call though as I am afraid I ask for too many changes. Could be a followup.

Reply:
Great suggestion! Addressed in the latest commit.